src/lib.rs

```diff
@@ -521,17 +487,86 @@ pub enum LocParseError {
     SecondTabulatorMissing,
 }
 
-/// A brief in-app documentation.
-///
-/// This function will write a brief usage information, including a short
-/// introduction to the meaning of the configured `options`, to `stdout`.
-pub fn usage(options: &getopts::Options) {
-    let description = "Parses the Git history.";
-    let name = "commit-analyzer";
-    let synopsis = "[OPTIONS]";
-
-    println!(
-        "{name}.\n{description}\n\n{}",
-        options.usage(&format!("Usage: {name} {synopsis}"))
-    );
+/// Parses the Git history.
+#[derive(Parser, Debug)]
+#[clap(version, about, long_about = None)]
+pub struct Args {
+    #[clap(subcommand)]
+    input_method: InputMethod,
+
+    /// Always show the entire output.
+    #[clap(short, long)]
+    verbose: bool,
+
+    /// Filter for certain author names. ORs if specified multiple times.
+    #[clap(short, long)]
+    author_contains: Vec<String>,
+
+    /// Filter for certain author names. ORs if specified multiple times.
+    #[clap(long)]
+    author_equals: Vec<String>,
+
+    /// Filter for certain author emails. ORs if specified multiple times.
+    #[clap(short, long)]
+    email_contains: Vec<String>,
+    /// Filter for certain author emails. ORs if specified multiple times.
+    #[clap(long)]
+    email_equals: Vec<String>,
+
+    /// Filter for certain commit hashes. ORs if specified multiple times.
+    #[clap(short, long)]
+    commit_contains: Vec<String>,
+    /// Filter for certain commit hashes. ORs if specified multiple times.
+    #[clap(long)]
+    commit_equals: Vec<String>,
+
+    /// Filter loc for certain file extension (e.g. `--file-extension cpp`). ORs if specified multiple times.
+    #[clap(short, long)]
+    file_extension: Vec<String>,
+
+    /// Filter for certain commit messages. ORs if specified multiple times.
+    #[clap(short, long)]
+    message_contains: Vec<String>,
+    /// Filter for certain commit messages. ORs if specified multiple times.
+    #[clap(long)]
+    message_equals: Vec<String>,
+    /// Filter for certain commit messages. ORs if specified multiple times.
+    #[clap(short = 'l', long)]
+    message_starts_with: Vec<String>,
+
+    /// The time which may pass between two commits that still counts as working.
+    #[clap(short, long, default_value_t = 3)]
+    duration: u32,
+
+    /// An output file for the commits per day in CSV format.
+    #[clap(short, long)]
+    output: Option<PathBuf>,
+}
+
+impl Args {
+    /// Get a reference to the input method specified by the user.
+    #[must_use]
+    pub fn input_method(&self) -> &InputMethod {
+        &self.input_method
+    }
+
+    /// Get the expected verbosity of the program.
+    #[must_use]
+    pub fn is_verbose(&self) -> bool {
+        self.verbose
```

Comment:
yeah, usually. Thing is, that imho bools should always be written with a verb like `is_`, but a parameter `--is-verbose` would be very ugly. But I check, how one can change the name of the flag.

Reply:
I am not such a big fan of these prefixes for variables -- like you use it in the first review comment --, they sound to me more like function / method prefixes; but this is surely a matter of taste.

The easiest thing would be to just rename the method to `verbose` since

```rust
if args.verbose() {
    // ...
}
```

is still of sufficient expressiveness to avoid misunderstandings and fixes the grammar issue arising from the current name.

Reply:
I decided to just rename the field to `is_verbose` because I personally think it is a way better way to name bool expressions.

Reply:
Okay. :+1: